src/lib/stitching/convection/v2/stitching.ts

```diff
@@ -90,5 +100,45 @@ export const consignmentStitchingEnvironment = (
           ),
       },
     },
+    Mutation: {
+      createConsignmentSubmission: {
+        resolve: async (_source, args, context, info) => {
+          const myCollectionArtworkID = args.input?.myCollectionArtworkID
+          let createSubmissionArgs = args
+
+          // when myCollectionArtworkID is specified, use artwork data to fill in submission
+          if (myCollectionArtworkID) {
+            if (context.artworkLoader) {
+              const artwork = await context.artworkLoader(myCollectionArtworkID)
+
+              if (!artwork) {
+                throw new Error("Artwork not found")
+              }
+
+              const artworkSubmissionData = artworkToSubmissionMapping(artwork)
+
+              // use artwork data to fill in submission, but allow input to override
+              createSubmissionArgs = {
+                ...createSubmissionArgs,
+                input: {
+                  ...artworkSubmissionData,
+                  ...createSubmissionArgs.input,
```

Comment:
I think as written, if we currently have clients which execute `createConsignmentSubmission` while passing both `myCollectionArtworkID` and other artwork data input arguments, any input argument with a value of `null` will override any fetched value from the artwork.

Might be fine for this iteration but something to double-check on in Force and Eigen.

```typescript
const artworkSubmissionData = { title: "Penicillin G Procaine" }
const input = { title: null }

const merged = {
  ... artworkSubmissionData,
  ...input,
}
// { title: null }
```

Reply:
In Eigen currently, all values would be overridden by the form inputs when submitting a MyC artwork, and in Force, we would create the submission from the MyC artwork without sending any additional (artwork-related) data.

I guess overriding fields with `null` values is okay. It would allow clearing fields in the submission form that have been set on the MyC artwork.